rest/route/user.go

```diff
@@ -387,13 +389,14 @@ func (h *userRolesPostHandler) Factory() gimlet.RouteHandler {
 }
 
 func (h *userRolesPostHandler) Parse(ctx context.Context, r *http.Request) error {
+	if err := utility.ReadJSON(r.Body, h); err != nil {
```

Comment:
Can you put roles/createUser in an options struct instead? It's bad form to read into the handler (also is better for documentation).

Reply:
done